test/Builder.php

```diff
@@ -0,0 +1,533 @@
+                        return $builder
+                            ->withId('biophysics-structural-biology')
+                            ->withName('Biophysics and Structural Biology')
+                            ->withPromiseOfImpactStatement('Subject biophysics-structural-biology impact statement');
+                    },
+                    'epidemiology-global-health' => function ($builder) {
+                        return $builder
+                            ->withId('epidemiology-global-health')
+                            ->withName('Epidemiology and Global Health')
+                            ->withPromiseOfImpactStatement('Subject epidemiology-global-health impact statement');
+                    },
+                    'microbiology-infectious-disease' => function ($builder) {
+                        return $builder
+                            ->withId('microbiology-infectious-disease')
+                            ->withName('Microbiology and Infectious Disease')
+                            ->withPromiseOfImpactStatement('Subject microbiology-infectious-disease impact statement');
+                    },
+                ],
+            ];
+        }
+
+        return self::$sampleRecipes;
+    }
+
+    public static function for($model) : self
+    {
+        return (new self())->create($model);
+    }
+
+    /**
+     * @return object instance of $model
+     */
+    public static function dummy($model)
+    {
+        return self::for($model)->__invoke();
+    }
+
+    public function create($model) : self
+    {
+        $this->model = $model;
+        $defaults = $this->defaultTestData($model);
+        if (!array_key_exists($model, $defaults)) {
+            throw new InvalidArgumentException("No defaults available for $model");
+        }
+
+        $this->testData = call_user_func($defaults[$model]);
+
+        return $this;
+    }
+
+    /**
+     * @method with...($value)  e.g. withImpactStatement('a string')
+     * @method withPromiseOf...($value)  e.g. withPromiseOfBanner(new Image(...))
+     */
+    public function __call($name, $args) : self
+    {
+        if (preg_match('/^withPromiseOf(.*)$/', $name, $matches)) {
+            $field = lcfirst($matches[1]);
+            $this->ensureExistingField($field);
+            $this->ensureSingleArgument($args);
+            $this->testData[$field] = promise_for($args[0]);
+        } elseif (preg_match('/^with(.*)$/', $name, $matches)) {
+            $field = lcfirst($matches[1]);
+            $this->ensureExistingField($field);
+            $this->ensureSingleArgument($args);
+            $this->testData[$field] = $args[0];
+        } else {
+            throw new BadMethodCallException("Magic method $name is not supported by the Builder class");
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return object instance of $this->model
+     */
+    public function __invoke()
+    {
+        $class = new \ReflectionClass($this->model);
```

Comment:
No, there's a bunch of these in this file already (eg `DateTimeImmutable` and `LogicException`).